### PR TITLE
kops-ci: Bump EKS module

### PR DESCRIPTION
**Pull request by ameukam**
Bump to v20.x version

Introducing:
- Access entries
- Karpenter support
- AL2023 support
- tags on launch templates instead of the node group

/hold

### CODE REVIEW

**Comment by ameukam** [comment]
cc @bryantbiggs 

I see this in the `terraform plan`:

```bash
  # module.eks.time_sleep.this[0] must be replaced
+/- resource "time_sleep" "this" {
      ~ id              = "2023-12-05T08:28:21Z" -> (known after apply)
      ~ triggers        = { # forces replacement
          + "cluster_service_cidr"               = "172.20.0.0/16"
            # (4 unchanged elements hidden)
        }
        # (1 unchanged attribute hidden)
    }
```

Is it expected ?

**Comment by bryantbiggs** [comment]
yes - see https://github.com/terraform-aws-modules/terraform-aws-eks/pull/2963

File: infra/aws/terraform/kops-infra-ci/eks.tf [hunk 1]
```diff
@@ -37,37 +41,25 @@ module "eks" {
...
-  ]
-
   cloudwatch_log_group_retention_in_days = 30
 
   cluster_addons = {
     coredns = {
-      most_recent       = true
-      resolve_conflicts = "OVERWRITE"
+      most_recent                 = true
+      resolve_conflicts_on_update = "OVERWRITE"
```
Comment (bryantbiggs) [inline infra/aws/terraform/kops-infra-ci/eks.tf:49]: fyi, on v20, all addons default to this value https://github.com/terraform-aws-modules/terraform-aws-eks/blob/1627231af669796669ce83e0a4672a7e6d94a0b3/main.tf#L503

File: infra/aws/terraform/kops-infra-ci/eks.tf [hunk 2]
```diff
@@ -17,14 +17,18 @@ limitations under the License.
...
 
   cluster_name                   = local.cluster_name
   cluster_version                = var.eks_version
   cluster_endpoint_public_access = true
 
   cluster_ip_family = "ipv4"
 
+  # Give the Terraform identity admin access to the cluster
+  # which will allow resources to be deployed into the cluster
+  enable_cluster_creator_admin_permissions = true
```
Comment (ameukam) [inline infra/aws/terraform/kops-infra-ci/eks.tf:30]: @bryantbiggs Do we still need the kubernetes provider with this ?

File: infra/aws/terraform/kops-infra-ci/eks.tf (hunk 1 above)
Comment (ameukam) [reply infra/aws/terraform/kops-infra-ci/eks.tf]: Dropped

File: infra/aws/terraform/kops-infra-ci/eks.tf (hunk 2 above)
Comment (bryantbiggs) [reply infra/aws/terraform/kops-infra-ci/eks.tf:30]: no - you only need the Kubernetes provider if you will continue to create `aws-auth` entries using the (new) [sub-module](https://github.com/terraform-aws-modules/terraform-aws-eks/tree/master/modules/aws-auth). This was the only area where the Kubernetes provider was used

Comment (ameukam) [reply infra/aws/terraform/kops-infra-ci/eks.tf:30]: I still have a few [resources](https://github.com/kubernetes/k8s.io/blob/main/infra/aws/terraform/kops-infra-ci/metrics-server.tf) created by the k8s provider. So If understand correctly, those resources can be actuated using the access entries ?

Comment (bryantbiggs) [reply infra/aws/terraform/kops-infra-ci/eks.tf:30]: I don't see any resources at that link that could be replaced by access entry. In short:
- Access entry allows IAM principals (roles and users) access to Kubernetes resources on the cluster
- Pod identity maps an IAM role to a service account in a namespace in the specified cluster

**Review by bryantbiggs — APPROVED** [review]
/lgtm 🚀

**Pull request merged**